compiler/src/iree/compiler/Codegen/Common/BlockDynamicDimensions.cpp

```diff
@@ -186,68 +195,104 @@ blockDynamicDimensionsOfValue(RewriterBase &rewriter,
 /// ```
 static LogicalResult blockDynamicDimensions(
     RewriterBase &rewriter, const TensorDynamicDimAnalysis &dynamicDimAnalysis,
-    Operation *operation, llvm::SmallDenseSet<int64_t> limitToOperandIndices) {
-  OpBuilder::InsertionGuard g(rewriter);
-
+    Operation *operation, llvm::SmallDenseSet<int64_t> limitToOperandNumbers,
+    llvm::SmallDenseSet<int64_t> limitToResultNumbers) {
   for (OpOperand &operand : operation->getOpOperands()) {
-    if (!limitToOperandIndices.contains(operand.getOperandNumber()))
+    if (!limitToOperandNumbers.contains(operand.getOperandNumber()))
       continue;
     if (operand.get().getDefiningOp<tensor::CollapseShapeOp>())
       continue;
     TensorDivisibilityInfo operandDivisibilityInfo =
         getTensorDivisibilityInfo(dynamicDimAnalysis, operand.get());
     if (operandDivisibilityInfo.empty())
       continue;
-    std::optional<Value> newOperand = blockDynamicDimensionsOfValue(
+    std::optional<ReshapeOps> reshapes = blockDynamicDimensionsOfValue(
         rewriter, operandDivisibilityInfo, operand.get());
-    if (newOperand) {
-      rewriter.modifyOpInPlace(operation,
-                               [&]() { operand.set(newOperand.value()); });
+    if (reshapes) {
+      rewriter.modifyOpInPlace(
+          operation, [&]() { operand.set(reshapes->collapseShapeOp); });
+    }
+  }
+
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPointAfter(operation);
+  for (OpResult result : operation->getResults()) {
+    if (!limitToResultNumbers.contains(result.getResultNumber()))
+      continue;
+    TensorDivisibilityInfo resultDivisibilityInfo =
+        getTensorDivisibilityInfo(dynamicDimAnalysis, result);
+    if (resultDivisibilityInfo.empty())
+      continue;
+    std::optional<ReshapeOps> reshapes =
+        blockDynamicDimensionsOfValue(rewriter, resultDivisibilityInfo, result);
+    if (reshapes) {
```

Comment:
NIT: can we do `if(!reshapes) continue` ?

Reply:
Sure. I am landing this now.. ill fix up since I am making some changes here now.